activesupport/lib/active_support/broadcast_logger.rb

```diff
@@ -231,6 +269,17 @@ def dispatch(&block)
         @broadcasts.each { |logger| block.call(logger) }
       end
 
+      def dispatch_with_processors(&block)
+        @broadcasts.each do |logger|
+          logger.extend(LogProcessor) unless logger.is_a?(LogProcessor)
```

Comment:
Can we extend the logger when creating the broadcast? Otherwise we will invalidate the method cache at runtime instead of boottime.

Reply:
Ah yes, good point 👍

Reply:
Moved this to the [`broadcast_to`](https://github.com/rails/rails/pull/49771/commits/ac23c0da581bc9511fde2bf636b828887cc4c747#diff-4923131d6bb7168dd7d0d38c819e847794e0a39c060d95da108f481978749fbdR132-R134 ) method (which gets called when creating the broadcast).
This is to support adding loggers to the broadcast after initialisation and keep the broadcasting functionality working.